clib/cList.mli

```diff
@@ -38,6 +38,9 @@ sig
   val for_all2eq : ('a -> 'b -> bool) -> 'a list -> 'b list -> bool
   (** Same as [List.for_all2] but returning [false] when of different length *)
 
+  val exists_i : (int -> 'a -> bool) -> int -> 'a list -> bool
```

Comment:
We currently have different styles for such iterators in `CList`:
- `for_all_i`, `map_i`, `map2_i`, `fold_right_i`, `fold_left_i` which have the `(int -> 'a -> ...) -> int -> 'a list -> ...` format
- `filteri`, `map_filter_i`, `partitioni`, which have the `(int -> 'a -> ...) -> 'a list -> ...` format, starting with 0
- `init`, which as the `int -> (int -> ...) -> ...` format (but also arguably of a slightly different nature)

In `CArray`, we have respectively:
- `for_all_i`
- `fold_right_i`, `fold_left_i`, `fold_left2_i`, `map2_i`, `map3_i`, `iter2_i`, `fold_left2_map_i`, `map_i`
- none

In `List` (OCaml's stdlib), we have:
- none
- `iteri`, `mapi`, `filteri` (bis)
- none

In `Array`, we have:
- none
- `iteri`, `mapi`
- `init`

I agree that we should unify the style (e.g. to follow OCaml's original style, up to the `i` vs `_i`). Here, my concern was local, i.e. to have the same format as `for_all_i` since they are going together.

Reply:
fair enough